src/main.cpp

```diff
@@ -40,22 +41,27 @@ int main(int argc, char **argv) {
   ros::Rate update_rate(update_rate_hz);
   ros::Time ts = ros::Time::now();
   while (!shouldShutdown) {
-    forqueSensorHW.update();
+    if (!forqueSensorHW.update()) {
+      errorShutdown = true;
+      break;
+    }
     ros::Duration d = ros::Time::now() - ts;
     ts = ros::Time::now();
     cm.update(ts, d);
     update_rate.sleep();
   }
 
   // wait for a bit, so controller_manager can unload controllers
-  ros::Duration shutdownDuration(1.5);
-  ros::Time shutdownBegin = ros::Time::now();
-  while (ros::Time::now() - shutdownBegin < shutdownDuration) {
-    ros::Duration d = ros::Time::now() - ts;
-    ts = ros::Time::now();
-    cm.update(ts, d);
+  if (!errorShutdown) {
```

Comment:
I wonder if it's clearer to make this conditioned on `shouldShutdown`, which should now only be true if we shutdown "nicely" via sigint.

Reply:
Also, *technically*, conditioning this on `shouldShutdown` could lead to a race condition where we `sigint` between the `break` and this if statement, leading to undesirable behavior (if we error, we need to shutdown *now*).

So, unlikely, but better to be safe.